scipy/stats/_binomtest.py

```diff
@@ -0,0 +1,314 @@
+            of the estimated proportion. Default is 0.95.
+        method : {'exact', 'wilson', 'wilsoncc'}, optional
+            Selects the method used to compute the confidence interval
+            for the estimate of the proportion:
+
+            'exact' :
+                Use the Clopper-Pearson exact method [1]_.
+            'wilson' :
+                Wilson's method, without continuity correction ([2]_, [3]_).
+            'wilsoncc' :
+                Wilson's method, with continuity correction ([2]_, [3]_).
+
+            Default is ``'exact'``.
+
+        Returns
+        -------
+        ci : namedtuple with fields ``low`` and ``high``
+            Contains the lower and upper bounds of the confidence interval.
+
+        References
+        ----------
+        .. [1] C. J. Clopper and E. S. Pearson, The use of confidence or
+               fiducial limits illustrated in the case of the binomial,
+               Biometrika, Vol. 26, No. 4, pp 404-413 (Dec. 1934).
+        .. [2] E. B. Wilson, Probable inference, the law of succession, and
+               statistical inference, J. Amer. Stat. Assoc., 22, pp 209-212
+               (1927).
+        .. [3] Robert G. Newcombe, Two-sided confidence intervals for the
+               single proportion: comparison of seven methods, Statistics
+               in Medicine, 17, pp 857-872 (1998).
+
+        Examples
+        --------
+        >>> from scipy.stats import binomtest
+        >>> result = binomtest(k=7, n=50, p=0.1)
+        >>> result.proportion_estimate
+        0.14
+        >>> result.proportion_ci
+        ConfidenceInterval(low=0.04246878737883377, high=0.30910696596481874)
+        """
+        if method not in ('exact', 'wilson', 'wilsoncc'):
+            raise ValueError("method must be one of 'exact', 'wilson' or "
+                             "'wilsoncc'.")
+        if not (0 <= confidence_level <= 1):
+            raise ValueError('confidence_level must be in the interval '
+                             '[0, 1].')
+        if method == 'exact':
+            low, high = _binom_exact_conf_int(self.k, self.n,
+                                              confidence_level,
+                                              self.alternative)
+        else:
+            # method is 'wilson' or 'wilsoncc'
+            low, high = _binom_wilson_conf_int(self.k, self.n,
+                                               confidence_level,
+                                               self.alternative,
+                                               correction=method == 'wilsoncc')
+        return ConfidenceInterval(low=low, high=high)
+
+
+def _binom_exact_conf_int(k, n, confidence_level, alternative):
+    """
+    Compute the estimate and confidence interval for the binomial test.
+
+    Returns proportion, prop_low, prop_high
+    """
+    if alternative == 'two-sided':
+        alpha = (1 - confidence_level) / 2
+        if k == 0:
+            plow = 0.0
```

Comment:
Most of these special cases are untested.

Reply:
I added unit tests for the edge cases.

I didn't address @rlucas7's comment, so don't resolve this comment.  I'm not sure the suggested change is better, but I'll take a look.

Reply:
@rlucas7, in one sense that would be simpler, but I prefer to read this code as a switch statement with three branches, where the job of each branch is to compute the confidence interval for the given alternative.  Assigning default values to `plow` and `phigh` in the beginning and then fixing the values depending on the alternative is conceptually more complicated. For example, if I'm reading the code for the `'greater'` alternative, I don't want to have to look around to see where `phigh` was set or where `plow` was set if k is 0.  Admittedly, it wouldn't be that bad, since the default values would be assigned just 20 or so lines back, but I prefer to keep the assignments as they are.

Reply:
@WarrenWeckesser ok, not going to push on this one, marking resolved